test/methods/gd_nesterov_accelerated.jl

```diff
@@ -0,0 +1,220 @@
+                0 #stop iteration 
+            )
+
+            ## Incorrect number of Iterates stored (does not check dimension)
+            @test_throws AssertionError NesterovAcceleratedGD{float_type}(
+                "NAGD",
+                float_type(1.0), #Step Size 
+                randn(float_type, param_dim), # z buffer 
+                randn(float_type, param_dim), # y buffer 
+                float_type(0), #B
+                float_type(1e-3), #threshold 
+                max_iter, #max_iterations 
+                Vector{float_type}[Vector{float_type}(undef, param_dim) for 
+                    i = 1:max_iter], #iterate storage 
+                randn(float_type, max_iter+1), #grad storage 
+                0 #stop iteration 
+            )
+
+            ## Incorrect number of entries for gradient norm storage
+            @test_throws AssertionError NesterovAcceleratedGD{float_type}(
+                "NAGD",
+                float_type(1.0), #Step Size 
+                randn(float_type, param_dim), # z buffer 
+                randn(float_type, param_dim), # y buffer 
+                float_type(0), #B
+                float_type(1e-3), #threshold 
+                max_iter, #max_iterations 
+                Vector{float_type}[Vector{float_type}(undef, param_dim) for 
+                    i = 1:max_iter+1], #iterate storage 
+                randn(float_type, max_iter), #grad storage 
+                0 #stop iteration 
+            )
+            
+            ## Mismatch in y and z buffer dimensions 
+            @test_throws AssertionError NesterovAcceleratedGD{float_type}(
+                "NAGD",
+                float_type(1.0), #Step Size 
+                randn(float_type, param_dim), # z buffer 
+                randn(float_type, param_dim-1), # y buffer 
+                float_type(0), #B
+                float_type(1e-3), #threshold 
+                max_iter, #max_iterations 
+                Vector{float_type}[Vector{float_type}(undef, param_dim) for 
+                    i = 1:max_iter+1], #iterate storage 
+                randn(float_type, max_iter+1), #grad storage 
+                0 #stop iteration 
+            )
+        end
+    end
+
+    ##########################################
+    # Test optimizer
+    ##########################################
+    progData = OptimizationMethods.LeastSquares(Float64)
+    x0 = progData.meta.x0 
+    step_size = 1e-3
+    threshold = 1e-8
+
+    # Base Case(s)
+    let progData=progData, x0=x0, step_size=step_size, threshold=threshold
+
+        optData = NesterovAcceleratedGD(
+            Float64,
+            x0 = x0,
+            step_size = step_size, 
+            threshold = threshold, 
+            max_iterations = 1
+        )
+
+        x1 = nesterov_accelerated_gd(optData, progData)
+
+        # Test updated value
+        @test optData.stop_iteration == 1
+        @test optData.iter_hist[2] == x1 
```

Comment:
We never test if `optData.iter_hist[1] = x0`, although this might imply that.

Reply:
It might be a good test anyway to make sure the method actually copies the first iterate from `optData.iter_hist`